Improve contest start times

deboer-tim opened the pull request:
When a contest start time is empty (returns null in json - not scheduled, or countdown paused) the output from the `contest` command is a little bizarre due to Go's nil type (and my not thinking about it before):
     finals: ICPC World Finals 2019
             5h0m0s starting at 0001-01-01 00:00:00 +0000 UTC

While fixing this I expanded it to include the following output:
     finals: ICPC World Finals 2019
             5h0m0s (not scheduled)
     finals: ICPC World Finals 2019
             5h0m0s (countdown paused at 5m34)
     finals: ICPC World Finals 2019
             5h0m0s starting at 2019-04-04 06:50:25 -0400 EDT
     finals: ICPC World Finals 2019
             Contest running. 5h0m0s started at 2019-04-04 06:50:25 -0400 EDT
     finals: ICPC World Finals 2019
             Contest over. Started at 2019-04-04 06:50:25 -0400 EDT

nickygerritsen reviewed (approved):
Probably if we add some `HumanFriendlyName()`  to ApiInteractor we can do the nice display thingy easily

deboer-tim committed:
Improve contest start times

When a contest start time is empty (returns null in json - not scheduled, or countdown paused) the output from the `contest` command is a little bizarre due to Go's nil type (and my not thinking about it before):
     finals: ICPC World Finals 2019
             5h0m0s starting at 0001-01-01 00:00:00 +0000 UTC

While fixing this I expanded it to include the following output:
     finals: ICPC World Finals 2019
             5h0m0s (not scheduled)
     finals: ICPC World Finals 2019
             5h0m0s (countdown paused at 5m34)
     finals: ICPC World Finals 2019
             5h0m0s starting at 2019-04-04 06:50:25 -0400 EDT
     finals: ICPC World Finals 2019
             Contest running. 5h0m0s started at 2019-04-04 06:50:25 -0400 EDT
     finals: ICPC World Finals 2019
             Contest over. Started at 2019-04-04 06:50:25 -0400 EDT

deboer-tim committed:
Update api-interactor dependency

deboer-tim committed:
Fix go.sum and tidy up
